lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -45,9 +57,98 @@
     seed_milestone_relationship_types,
 )
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 # pylint: disable=protected-access
 
 
+class CoachAccessTestCaseCCX(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Test if user is coach on ccx.
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up course for tests
+        """
+        super(CoachAccessTestCaseCCX, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(CoachAccessTestCaseCCX, self).setUp()
+
+        # Create ccx coach account
+        self.coach = AdminFactory.create()
+        self.client.login(username=self.coach.username, password="test")
```

Comment:
@singingwolfboy  please see code below:

[AdminFactory](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/tests/factories.py#L114) is child class of [UserFactory](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/tests/factories.py#L71). And inside ``UserFactory`` we have [hard coded password test](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/tests/factories.py#L78)

> password = factory.PostGenerationMethodCall('set_password', 'test')